packages/core/upload/admin/src/constants.js

```diff
@@ -34,12 +34,15 @@ export const FolderDefinition = PropTypes.shape({
     count: PropTypes.number.isRequired,
   }),
   name: PropTypes.string.isRequired,
-  parent: PropTypes.oneOfType([ParentFolderDefinition, PropTypes.number]),
   updatedAt: PropTypes.string.isRequired,
   updatedBy: PropTypes.shape(),
   pathId: PropTypes.number.isRequired,
   path: PropTypes.string.isRequired,
-});
+};
+
+FolderShape.parent = PropTypes.shape(ParentFolderShape);
```

Comment:
@joshuaellis I just can not make

```
PropTypes.oneOfType([ParentFolderDefinition, PropTypes.number]),
```

work - it will always expect a `Number`. Do you have an idea?

Reply:
Alright, thanks. I think I'd prefer to leave it as it is now if that works for you and to add this case to the definition of `FolderShape` once it is needed.

Reply:
Works for me 😄